### ISSUE TITLE
useNuxtApp() and useRuntimeConfig not available in pinia store during SSR

### ISSUE THREAD

**Issue by stephenjason89** [nuxt/bridge#467]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v18.7.0`
- Nuxt Version:     `2.16.0-27616340.013f051b`
- Package Manager:  `yarn@3.2.1`
- Builder:          `webpack`
- User Config:      `buildDir`, `srcDir`, `bridge`, `components`, `ssr`, `dev`, `alias`, `image`, `generate`, `head`, `typescript`, `publicRuntimeConfig`, `privateRuntimeConfig`, `css`, `loading`, `loadingIndicator`, `plugins`, `buildModules`, `modules`, `proxy`, `axios`, `vuetify`, `build`, `serverHandlers`, `devServerHandlers`
- Runtime Modules:  `@nuxtjs/vuetify@1.12.3`, `@nuxt/image@0.7.1`, `@nuxtjs/axios@5.13.6`, `@nuxtjs/proxy@2.1.0`, `nuxt-social-meta@1.0.0`
- Build Modules:    `()`, `nuxt-storm@1.1.3`, `nuxt-compress@5.0.0`, `@pinia/nuxt@0.3.1`, `~/modules/apolloSSR`, `@nuxt/bridge@3.0.0-27639120.e7a7f9b`
------------------------------

### Reproduction

https://github.com/vuejs/pinia/discussions/1526

### Describe the bug

I am getting this error
<img width="584" alt="image" src="https://user-images.githubusercontent.com/3810177/183127061-cc6add47-9ea8-4b8e-9ebc-2aa9cb43d3c8.png">

When running this pinia actions during SSR
<img width="857" alt="image" src="https://user-images.githubusercontent.com/3810177/183127139-e2160804-bbc2-4d0d-aae4-ae96e08d3de9.png">

I rolled back to the old this.$nuxt for now to temporarily solve the issue

<img width="861" alt="image" src="https://user-images.githubusercontent.com/3810177/183127200-92f3cbaa-264c-405b-8bf0-03a223542a09.png">

### Additional context

The page being loaded uses the default.vue layout which has an apollo query "labels"

<img width="684" alt="image" src="https://user-images.githubusercontent.com/3810177/183127475-56c43f92-d5c1-4661-a56d-cc571d0f1e3f.png">

So that, together with this query causes the error.
<img width="288" alt="image" src="https://user-images.githubusercontent.com/3810177/183127649-ecf13771-2433-4ae7-b194-7ecc8807cec9.png">

_No response_

**Comment by stephenjason89** [comment]
I think this might be related, @danielroe @antlionguard

https://github.com/nuxt/bridge/issues/436

The reproduction needs to have SSR:true in order to see the error

**Comment by Lanayx** [comment]
I experience this as well
https://github.com/vuejs/pinia/discussions/1513

**Comment by stephenjason89** [comment]
@Lanayx for the meantime this.$nuxt helped me get around this problem

**Comment by Lanayx** [comment]
@stephenjason89 this.$nuxt is unavailable in latest pinia :( https://github.com/vuejs/pinia/discussions/1526

**Comment by antlionguard** [comment]
> @stephenjason89 this.$nuxt is unavailable in latest pinia :( [vuejs/pinia#1526](https://github.com/vuejs/pinia/discussions/1526)

It's available now but not have a type support.

**Comment by stephenjason89** [comment]
> @stephenjason89 this.$nuxt is unavailable in latest pinia :( [vuejs/pinia#1526](https://github.com/vuejs/pinia/discussions/1526)

Just no type support. 
Use @ts-ignore

**Comment by Lanayx** [comment]
@antlionguard can you please give more information - where it is availabe? 
I'm using 
"@pinia/nuxt": "^0.4.0",
"nuxt": "3.0.0-rc.8",
"pinia": "^2.0.18"
And `this.$nuxt` is undefined inside store action on server side

**Comment by Lanayx** [comment]
Ah, I see, it was removed from Nuxt3, so Nuxt2 users are more lucky here

**Comment by stephenjason89** [comment]
Let's hope it gets fixed soon 😅 @Lanayx 
I'm using nuxt bridge

**Comment by Lanayx** [comment]
As far as I can understand Nuxt3 developers motivation - they want us to pass nuxtApp explicitly into actions parameters, rather than implicitly receiving it from `this`

**Comment by stephenjason89** [comment]
@Lanayx That's correct. I hope it gets fixed soon

**Comment by Luferov** [comment]
I can't continue development anymore because of this error. It is sad 😭

**Comment by Luferov** [comment]
maybe the problem is not with nuxt/bridge btw. I installed nuxt/apollo modules first and everything broke, but when I moved it to the end it worked

**Comment by LudoFont** [comment]
up please, is there any hope for this bug to be fixed ? (nuxt3 + pinia).
This bug happen to me when I call async methods in pinia actions during SSR.

**Comment by tubstrr** [comment]
I struggled with this for a while, the solution for me was to instantiate the store before I called a method inside of it. 

In my use case, I was trying to build Bread Crumbs aided by the current Route, my code looked like this:

```
export const useBreadCrumbStore = defineStore("breadcrumbs", () => {

  const { currentRoute } = useRouter();

  const makeBreadcrumbList = () => {
    const fullPath = currentRoute.value.fullPath;
    
    fullPath.split("/").forEach((path, index) => {
     ...
    }
  }
}
``` 

Which would error out with the `Nuxt Instance Unavailable`.

What I ended up having to do, was to call the store on my page, before my `useSEO` composable which actually calls the "makeBreadcumbList" function.   

I found this doc on Pinia's site which clarified what was happening in my case.  
https://pinia.vuejs.org/core-concepts/outside-component-usage.html

Hope this helps.
